checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1708 @@
+   * annotation whose target is the object whose field is being re-assigned.
+   *
+   * @param node an assignment node whose lhs is a non-final, owning field
+   * @param enclosingMethod the MethodTree in which the re-assignment takes place
+   */
+  private void checkEnclosingMethodIsCreatesMustCallFor(
+      AssignmentNode node, MethodTree enclosingMethod) {
+    Node lhs = node.getTarget();
+    if (!(lhs instanceof FieldAccessNode)) {
+      return;
+    }
+
+    String receiverString = receiverAsString((FieldAccessNode) lhs);
+    if (TreeUtils.isConstructor(enclosingMethod)) {
+      // Constructors always create must-call obligations, so there is no need for them to
+      // be annotated.
+      return;
+    }
+    ExecutableElement enclosingMethodElt = TreeUtils.elementFromDeclaration(enclosingMethod);
+    MustCallAnnotatedTypeFactory mcAtf =
+        typeFactory.getTypeFactoryOfSubchecker(MustCallChecker.class);
+
+    List<String> cmcfValues =
+        ResourceLeakVisitor.getCreatesMustCallForValues(enclosingMethodElt, mcAtf, typeFactory);
+
+    if (cmcfValues.isEmpty()) {
+      checker.reportError(
+          enclosingMethod,
+          "missing.creates.mustcall.for",
+          receiverString,
+          ((FieldAccessNode) lhs).getFieldName());
+      return;
+    }
+
+    String checked = "";
+    for (String targetStrWithoutAdaptation : cmcfValues) {
+      String targetStr = null;
+      try {
+        targetStr =
+            StringToJavaExpression.atMethodBody(
+                    targetStrWithoutAdaptation, enclosingMethod, checker)
+                .toString();
+      } catch (JavaExpressionParseException e) {
+        targetStr = targetStrWithoutAdaptation;
+      }
+      if (targetStr.equals(receiverString)) {
+        // This @CreatesMustCallFor annotation matches.
+        return;
+      }
+      if ("".equals(checked)) {
+        checked += targetStr;
+      } else {
+        checked += ", " + targetStr;
+      }
+    }
+    checker.reportError(
+        enclosingMethod,
+        "incompatible.creates.mustcall.for",
+        receiverString,
+        ((FieldAccessNode) lhs).getFieldName(),
+        checked);
+  }
+
+  /**
+   * Gets a standardized name for an object whose field is being re-assigned.
+   *
+   * @param fieldAccessNode a field access node
+   * @return the name of the object whose field is being accessed (the receiver), as a string
+   */
+  private String receiverAsString(FieldAccessNode fieldAccessNode) {
+    Node receiver = fieldAccessNode.getReceiver();
+    if (receiver instanceof ThisNode) {
+      return "this";
+    }
+    if (receiver instanceof LocalVariableNode) {
+
+      return ((LocalVariableNode) receiver).getName();
+    }
+    throw new BugInCF(
+        "unexpected receiver of field assignment: " + receiver + " of type " + receiver.getClass());
+  }
+
+  /**
+   * Finds the actual parameter passed in the {@code @MustCallAlias} position for a call.
```

Comment:
There is no such thing as an "actual parameter"; that term is a non sequitur.
There are "actual arguments" and "formal parameters".  I think this is trying to say "argument".  Please fix throughout.